src/dwi/tractography/seeding/basic.h

```diff
@@ -160,6 +161,119 @@ namespace MR
 
 
 
+        class Coordinate_parser
+        { MEMALIGN(Coordinate_parser)
+
+          public:
+
+            Eigen::MatrixXf coords;
+            size_t nr;
+            size_t nc;         
+
+            Coordinate_parser (const std::string& cds_path) {
+              
+              coords = load_matrix<float> (cds_path);
+              nr = coords.rows();
+              nc = coords.cols();
+
+            }  
+
+        };
+
+
+        
+        class Coordinates_fixed : public Base, public Coordinate_parser
+        { MEMALIGN(Coordinates_fixed)
+
+          public:
+
+            Coordinates_fixed (const std::string& in, const size_t n_streamlines) :
+              Base (in, "coordinate seeding fixed", MAX_TRACKING_SEED_ATTEMPTS_FIXED), 
+              Coordinate_parser(in) {
+
+                if (nc != 3)
+                  throw Exception ("Number of columns in \"" + in + "\" must equal 3!");   
+                if (n_streamlines == 0)
+                  throw Exception ("Number of streamlines to be generated per coordinate must be a positive integer!");             
+                current_coord = 0;
+                num_at_coord = 0;
+                expired = false;
+                nsl = n_streamlines;
+                count = nr * nsl;
```

Comment:
These can be set as initialisers for the constructor, rather than within the code body of the constructor.

Reply:
Could not figure out a way to set count as initialiser - it belongs to Base class and I could not access it from here, am I missing something? The rest have been set as initialisers now.